src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,296 @@
+        /// returns an <see cref="ExecutionResult"/>.
+        /// <br/><br/>
+        /// Override this method to mutate <see cref="ExecutionContext"/> as necessary, such
+        /// as changing the <see cref="ExecutionContext.RequestServices"/> property to a scoped instance.
+        /// </summary>
+        protected virtual async Task<ExecutionResult> ProcessDataAsync(ExecutionContext context, ExecutionNode node, object? value)
+        {
+            var result = new ExecutionResult(context);
+
+            try
+            {
+                var executionNode = BuildSubscriptionExecutionNode(node.Parent!, node.GraphType!, node.Field, node.FieldDefinition, node.IndexInParentNode, value!);
+
+                if (context.Listeners != null)
+                {
+                    foreach (var listener in context.Listeners)
+                    {
+                        await listener.BeforeExecutionAsync(context)
+                            .ConfigureAwait(false);
+                    }
+                }
+
+                // Execute the whole execution tree and return the result
+                await ExecuteNodeTreeAsync(context, executionNode).ConfigureAwait(false);
+
+                if (context.Listeners != null)
+                {
+                    foreach (var listener in context.Listeners)
+                    {
+                        await listener.AfterExecutionAsync(context)
+                            .ConfigureAwait(false);
+                    }
+                }
+
+                // Set the execution node's value to null if necessary
+                var dataIsNull = executionNode.PropagateNull();
+
+                // Return the result
+                result.Executed = true;
+                if (!dataIsNull || executionNode.FieldDefinition.ResolvedType is not NonNullGraphType)
+                {
+                    result.Data = new RootExecutionNode(null!, null)
+                    {
+                        SubFields = new ExecutionNode[]
+                        {
+                            executionNode,
+                        }
+                    };
+                }
+            }
+            catch (OperationCanceledException) when (context.CancellationToken.IsCancellationRequested)
+            {
+                // generate a static error message here;
+                // do not "throw;" back to the caller (the event source)
+                result.AddError(GenerateError(context, node, "The operation was canceled."));
+            }
+            catch (ExecutionError executionError)
+            {
+                executionError.Path = node.ResponsePath;
+                executionError.AddLocation(node.Field, context.Document);
+                result.AddError(executionError);
+            }
+            catch (Exception exception)
+            {
+                result.AddError(await HandleExceptionInternalAsync(context, node, exception,
```

Comment:
I see that `HandleExceptionInternalAsync` is called from 3 places but only this one _may_ pass `value` object among other parameters to `virtual HandleExceptionAsync` that _may be_ interested in it. What do you think?

Reply:
I'll have to open the code and read to understand what you're saying.

Reply:
`value` should be stored in `node.Source`.  I have reverted the change from above, moving the `var executionNode = BuildSubscriptionExecutionNode` back up near the top, and then using `node =` to overwrite the `node` variable and in so doing allow it to be picked up by the exception handling code.  There is no practical difference between the two instances to the error code, apart from the fact that `node.Source` could be accessed if access to `value` was desired.

Reply:
Done as described.

Reply:
I see, so `HandleExceptionAsync` may use `node.Source` now, right? And what is the value of `Source` property of "initial" node?

Reply:
I suppose `null`.

Reply:
> I see, so HandleExceptionAsync may use node.Source now, right? 

Yes

> And what is the value of Source property of "initial" node?  I suppose `null`.

I suppose so.  Do you have any further questions here?

Reply:
no, thanks